arch/ceva/src/common/up_createstack.c

```diff
@@ -233,6 +233,11 @@ void up_stack_color(FAR void *stackbase, size_t nbytes)
   uint32_t *stkptr = stackbase;
   size_t    nwords = nbytes / sizeof(uint32_t);
 
+  if (nwords == 0)
+    {
+      nwords = (uint32_t *)up_getsp() - stkptr;
+    }
+
```

Comment:
Can we somehow unify code for `up_stack_color` across the platforms? I see the API that implement the same with a different code:
```
void up_stack_color(FAR void *stackbase, size_t nbytes)
{
  uintptr_t start;
  uintptr_t end;
  size_t nwords;
  FAR uint32_t *ptr;

  /* Take extra care that we do not write outside the stack boundaries */

  start = INT32_ALIGN_UP((uintptr_t)stackbase);
  end   = nbytes ? INT32_ALIGN_DOWN((uintptr_t)stackbase + nbytes) :
          up_getsp(); /* 0: colorize the running stack */

  /* Get the adjusted size based on the top and bottom of the stack */

  nwords = (end - start) >> 2;
  ptr  = (FAR uint32_t *)start;

  /* Set the entire stack to the coloration value */

  while (nwords-- > 0)
    {
      *ptr++ = STACK_COLOR;
    }
}
```
and
```
void up_stack_color(FAR void *stackbase, size_t nbytes)
{
  /* Take extra care that we do not write outsize the stack boundaries */

  uint32_t *stkptr = (uint32_t *)(((uintptr_t)stackbase + 3) & ~3);
  uintptr_t stkend = nbytes ? (((uintptr_t)stackbase + nbytes) & ~3) :
                     up_getsp(); /* 0: colorize the running stack */
  size_t    nwords = (stkend - (uintptr_t)stackbase) >> 2;

  /* Set the entire stack to the coloration value */

  while (nwords-- > 0)
    {
      *stkptr++ = STACK_COLOR;
    }
}
```

Reply:
Done.